source/Ship.cpp

```diff
@@ -3467,9 +3467,14 @@ bool Ship::CanCarry(const Ship &ship) const
 	for(const auto &it : escorts)
 	{
 		auto escort = it.lock();
-		if(escort && escort.get() != &ship && escort->attributes.Category() == category
-			&& !escort->IsDestroyed())
+		if(!escort)
+			continue;
+		if(escort == ship.shared_from_this())
+			break;
```

Comment:
Why the break? If for some reason a ship has itself as an escort it should be ignored and the search should continue no?